tests/integration/pilot/protocolsniffing/main_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package protocolsniffing
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"istio.io/pkg/log"
+
+	"istio.io/istio/pkg/test/framework/resource"
+
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+var (
+	i istio.Instance
+	g galley.Instance
+	p pilot.Instance
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("protocolsniffing_test", m).
+		SetupOnEnv(environment.Kube, istio.Setup(&i, nil)).
+		Setup(func(ctx resource.Context) (err error) {
+			if g, err = galley.New(ctx, galley.Config{}); err != nil {
+				return err
+			}
+			if p, err = pilot.New(ctx, pilot.Config{
+				Galley: g,
+			}); err != nil {
+				return err
+			}
+			return nil
+		}).
+		Run()
+}
+
+func TestOutbound(t *testing.T) {
+	framework.NewTest(t).Run(func(ctx framework.TestContext) {
+		doTest(t, ctx)
+	})
+}
+
+func doTest(t *testing.T, ctx framework.TestContext) {
+	ns := namespace.NewOrFail(t, ctx, namespace.Config{
+		Prefix: "protocolsniffing",
+		Inject: true,
+	})
+
+	ports := []echo.Port{
+		{
+			Name:     "foo",
+			Protocol: protocol.HTTP,
+		},
+		{
+			Name:     "http",
+			Protocol: protocol.HTTP,
+		},
+		{
+			Name:     "bar",
+			Protocol: protocol.TCP,
+		},
+		{
+			Name:     "tcp",
+			Protocol: protocol.TCP,
+		},
+		{
+			Name:     "baz",
+			Protocol: protocol.GRPC,
```

Comment:
does this really test sniffing if we are defining a protocol? not sure how this maps to the envoy config

Reply:
This actually creates a port with name "baz". You can check `portgen.go`.